app/controllers/ops_controller/diagnostics.rb

```diff
@@ -92,13 +92,18 @@ def log_depot_edit
       end
 
       begin
-        if params[:log_protocol].blank? || params[:log_protocol] == "<No Depot>"
+        if params[:log_protocol].blank?
           @record.log_file_depot.try(:destroy)
         else
           new_uri = "#{params[:uri_prefix]}://#{params[:uri]}"
+          raise _("Unsupported log depot protocol: %{protocol}") % {:protocol => params[:log_protocol]} unless FileDepot.supported_depots.key?(params[:log_protocol])
```

Comment:
> `FileDepot.supported_depots.key?(params[:log_protocol])`

can `params[:log_protocol]` ever be a Class instance?

Because supported_depots returns a hash of `Klass => "display_name"`